pygmt/datatypes/grid.py

```diff
@@ -101,5 +103,128 @@ class _GMT_GRID_HEADER(ctp.Structure):  # noqa: N801
+    "gmtnc_grd_info" for reference.
+
+    The last dimension is special and is the data variable name, and the attributes
+    for this dimension are global attributes for the grid.
+
+    The grid is assumed to be Cartesian by default. If the x and y units are
+    "degrees_east" and "degrees_north", respectively, then the grid is assumed to be
+    geographic.
+
+    Parameters
+    ----------
+    header
+        The grid header structure.
+
+    Returns
+    -------
+    dims : tuple
+        The dimension names with the last dimension for the data variable.
+    attrs : dict
+        The attributes for each dimension.
+    registration : int
+        The grid registration. 0 for gridline and 1 for pixel.
+    gtype : int
+        The grid type. 0 for Cartesian grid and 1 for geographic grid.
+    """
+    # Default dimension names. The last dimension is for the data variable.
+    dims: tuple = ("x", "y", "z")
+    nameunits = (header.x_units, header.y_units, header.z_units)
+
+    # Dictionary for dimension attributes with the dimension name as the key.
+    attrs: dict = {dim: {} for dim in dims}
+    # Dictionary for mapping the default dimension names to the actual names.
+    newdims = {dim: dim for dim in dims}
+    # Loop over dimensions and get the dimension name and attributes from header
+    for dim, nameunit in zip(dims, nameunits, strict=False):
+        # The long_name and units attributes.
+        long_name, units = _parse_nameunits(nameunit.decode())
+        if long_name:
+            attrs[dim]["long_name"] = long_name
+        if units:
+            attrs[dim]["units"] = units
+
+        # "degrees_east"/"degrees_north" are the units for geographic coordinates
+        # following CF-conventions
+        if units == "degrees_east":
+            attrs[dim]["standard_name"] = "longitude"
+            newdims[dim] = "lon"
+        elif units == "degrees_north":
+            attrs[dim]["standard_name"] = "latitude"
+            newdims[dim] = "lat"
+
+        # Axis attributes are "X"/"Y"/"Z"/"T" for horizontal/vertical/time axis.
+        # The codes here may not work for 3-D grids.
+        if dim == dims[-1]:  # The last dimension is the data.
+            attrs[dim]["actual_range"] = np.array([header.z_min, header.z_max])
+        else:
+            attrs[dim]["axis"] = dim.upper()
+            idx = dims.index(dim) * 2
+            attrs[dim]["actual_range"] = np.array(header.wesn[idx : idx + 2])
+
+    # Cartesian or Geographic grid
+    gtype = 0
+    if (
+        attrs[dims[0]].get("standard_name") == "longitude"
+        and attrs[dims[1]].get("standard_name") == "latitude"
+    ):
+        gtype = 1
+    # Registration
+    registration = header.registration
+
+    # Update the attributes dictionary with new dimension names as keys
+    attrs = {newdims[dim]: attrs[dim] for dim in dims}
+    # Update the dimension names
+    dims = tuple(newdims[dim] for dim in dims)
+    return dims, attrs, registration, gtype
```

Comment:
I'm debating about the order of dimension names. Currently, the returned `dims` are `("x", "y", "z")`, but when creating a xarray.DataArray, we need to use `("y", "x", "z")` instead. Which one the `_parse_header` function should return?

Reply:
For multi-band images, the convention is to use `("z", "y", "x")` (band/channel, height, width), see e.g. rioxarray's example at https://corteva.github.io/rioxarray/stable/examples/COG.html. But we would then need to transpose the dims here.

Reply:
Oh wait, the `z` here refers to the raster pixel value (e.g. elevation), not the band. In that case, we can probably go with `("y", "x", "z")` order.